packages/shared/src/domain/address/zip.ts

```diff
@@ -34,3 +34,27 @@ export function normalizeZipCodeSafe(zipCode: string, strict = false): string |
   if (trimmedZip.trim().length === 8) return trimmedZip.slice(0, 4);
   return trimmedZip.slice(0, 5);
 }
+
+// TODO 2330 Move/merge this to normalizeZipCode
+export function normalizeZipCodeNew(
+  zipCode: string,
+  normalizeFn = normalizeZipCodeNewSafe
+): string {
+  const zipOrUndefined = normalizeFn(zipCode);
+  if (!zipOrUndefined) throw new Error("Invalid Zip.");
+  return zipOrUndefined;
+}
+
+// TODO 2330 Move/merge this to normalizeZipCodeSafe
+// TODO Should prob look into something that indicates ranges of possible values, like https://www.fincen.gov/sites/default/files/shared/us_state_territory_zip_codes.pdf
+export function normalizeZipCodeNewSafe(zipCode: string): string | undefined {
+  const trimmedZip = zipCode.trim();
+  if (trimmedZip === "") return undefined;
+  if (!trimmedZip.match(/^[0-9-]+$/)) return undefined;
+  if (trimmedZip.includes("-") && trimmedZip.split("-").length !== 2) return undefined;
+  const mainPart = trimmedZip.split("-")[0];
+  if (!mainPart) return trimmedZip;
+  if (mainPart.length === 5) return mainPart;
+  if (mainPart.length < 5) return mainPart.padStart(5, "0");
```

Comment:
I'd mark as invalid after < 4 -- zero padding 3/4 zeros seems less preferable than asking for input adjustment. Feels like guessing at that point.

Reply:
done

Reply:
3 digit zips are valid in NY.

Updated to require at least 3 digits.